packages/flame/lib/src/experimental/viewfinder.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../components/component.dart';
+import '../game/transform2d.dart';
+import 'camera.dart';
+import 'viewport.dart';
+
+/// [Viewfinder] is a part of a [Camera2] system that controls which part of
+/// the game world is currently visible through a viewport.
+///
+/// The viewfinder contains the game point that is currently at the
+/// "cross-hairs" of the viewport ([position]), the [zoom] level, and the
+/// [angle] of rotation of the camera.
+///
+class Viewfinder extends Component {
```

Comment:
I think a very useful feature of the current impl is the "follow" behaviour (including boundaries). how easy would it be to hook that up onto the viewfinder? might be even worth an example and/or helper methods

Reply:
Yes, that would be the next step (can't fit everything into one PR!)
I want effects system to recognize the `Viewfinder` and `Viewport` classes, and then you can attach various move/rotate/scale/shake/whatever effects to either of them. "Follow" and "respect boundary" aren't effects though, but they can be implemented as independent components. In the end it should have a greater set of capabilities than the original `Camera`.